connector/dojot_acl/app/kafka/KafkaConsumer.js

```diff
@@ -0,0 +1,113 @@
+const { ConfigManager, Logger, Kafka: { Consumer } } = require('@dojot/microservice-sdk');
+
+const logger = new Logger('dojot-acl:kafka-consumer');
+
+const DOJOT_ACL_CONFIG_LABEL = 'DOJOT_ACL';
+
+/**
+ * Consume kafka messages from kafka topics
+ */
+class kafkaConsumer {
+  /**
+     * Instance KafkaConsumer
+     */
+  constructor() {
+    this.config = ConfigManager.getConfig(DOJOT_ACL_CONFIG_LABEL);
+
+    this.consumerConfig = {
+      'kafka.consumer': { ...this.config.consumer },
+      'kafka.topic': { ...this.config.topic },
+    };
+
+    this.consumer = new Consumer(this.consumerConfig);
+    this.registeredCallbacks = new Map();
+  }
+
+  /**
+     * Initialize the kafka consumer
+     */
+  async init() {
+    try {
+      logger.info('init: kafka starting...');
+      await this.consumer.init();
+      logger.info('init: kafka started...');
+    } catch (error) {
+      logger.error(`init: Error starting kafka ${error.stack}`);
+      throw error;
+    }
+  }
+
+  /**
+     * Register callback for a topic
+     *
+     * @param {string} kafkaTopic
+     * @param {function} callback
+     */
+  registerCallback(kafkaTopic, callback) {
+    if (!this.registeredCallbacks.has(kafkaTopic)) {
+      logger.debug(`registerCallback: Register Callback for topic ${kafkaTopic}`);
+      const callbackId = this.consumer.registerCallback(kafkaTopic, callback);
+      this.registeredCallbacks.set(kafkaTopic, callbackId);
+    } else {
+      throw new Error(`registerCallback: Callback for topic ${kafkaTopic} already exist`);
+    }
+
+    logger.debug(`registerCallback: All Registered Callbacks ${[...this.registeredCallbacks]}`);
+  }
+
+  /**
+     * Unregister Callback for a given topic
+     *
+     * @param {string} kafkaTopic
+     */
+  unregisterCallback(kafkaTopic) {
+    if (this.registeredCallbacks.has(kafkaTopic)) {
+      logger.debug(`unregisterCallbacks: Unregister Callback for topic ${kafkaTopic}`);
+      const callbackId = this.registeredCallbacks.get(kafkaTopic);
+      this.consumer.unregisterCallback(callbackId);
+      this.registeredCallbacks.delete(kafkaTopic);
+    } else {
+      throw new Error(`unregisterCallbacks: Doesn't exist Callback to unregister for topic ${kafkaTopic}`);
+    }
+  }
+
+  /**
+     * Unregister all callbacks
+     */
+  unregisterAllCallbacks() {
+    logger.debug('unregisterCallbacks: Unregister All Callbacks');
+
+    this.registeredCallbacks.forEach((callbackId) => this.consumer.unregisterCallback(callbackId));
+    this.registeredCallbacks.clear();
+  }
```

Comment:
What is the need for these methods, it seems to me that they are not used anywhere ...

Reply:
could be used in graceful shutdown

Reply:
could be used in graceful shutdown

Reply:
Is it necessary to call this at shutdown? 
I saw that he already calls method `this.consumer.finish()` on shutdown process, which in turn performs the unsubscription of all topics... 
This is a question of mine too, I don't know the library well.

Reply:
I think not... 
Before the finish was not implemented.